src/ngAnimate/animateQueue.js

```diff
@@ -362,11 +359,19 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
         structural: isStructural,
         element: element,
         event: event,
+        addClass: options.addClass,
+        removeClass: options.removeClass,
         close: close,
         options: options,
         runner: runner
       };
 
+      if (!hasExistingAnimation) {
+        // normalization in this case means that it removes redundant CSS classes that
+        // already exist (addClass) or do not exist (removeClass) on the element
+        normalizeAnimationDetails(element, newAnimation);
+      }
+
       if (hasExistingAnimation) {
```

Comment:
This is fixed in the updated PR: https://github.com/angular/angular.js/pull/13822